app/models/user.rb

```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :comments
   has_many :likes
 
+  validates :name, presence: true
+  validates :posts_counter, comparison: { greater_than_or_equal_to: 0 }, presence: true
+
   def three_recent_user
     User.limit(3).order(created_at: :asc)
```

Comment:
- [ ] There is an `N+1` issue occurring while fetching the users' posts and their comments. You can use [bullet gem](https://github.com/flyerhzm/bullet) to easily detect N+1 issues.

![Screenshot from 2022-01-28 00-19-13](https://user-images.githubusercontent.com/1642014/151445343-adb225f1-97aa-41b6-ba35-6fe55ef52002.png)


From the project specification :arrow_down:  
> Make sure that the N+1 problem is solved when fetching all posts and their comments for a user

Reply:
Done! 👍